src/adapter/src/session/vars.rs

```diff
@@ -994,14 +1015,22 @@ pub struct SystemVars {
     max_roles: SystemVar<u32>,
     max_result_size: SystemVar<u32>,
     allowed_cluster_replica_sizes: SystemVar<Vec<String>>, // TODO: BTreeSet<String> will be better
+
+    // features
     window_functions: SystemVar<bool>,
-    config_has_synced_once: SystemVar<bool>,
+
+    // persist configuration
+    persist_blob_target_size: SystemVar<Option<usize>>,
```

Comment:
looks like most things here are not Options, did we do it here so that it's an override of the persist default? if so, maybe we're rather plumb the default here somehow and make it work like the others

(this is not a suggestion for you to have to pick off that refactor. if we do decide to go this directly, i'm happy to put anything in the two "None" spots above here and then I'll fill them in when I circle back to #16797)

Reply:
I had been thinking about making the persist config defaults associated consts on `PersistConfig` and then using those both in `PersistConfig::new()` and in the `ServerVar` definitions. The main thing that kept me back was that I wasn't sure if later changes to those defaults would apply to environments that hadn't explicitly overwritten the default. I thought that maybe we would just store the default `ServerVar` value in the stash and later have no way to say whether those values were defaults or explicitly overwritten by the user.

Of course I could just have tested this! I have now, and it looks like adapter does the right thing here, i.e. only store parameters in the stash that the user has `ALTER SYSTEM SET`, and remove them again on `ALTER SYSTEM RESET`.

So I agree, we should remove those `Option`s. Does having the defaults as associated consts on `PersistConfig` sound reasonable to you?

Reply:
I made the persist parameters non-optional now. LMK what you think!

Reply:
yeah, associated consts is perfect. I'm glad you were thinking about what's being stored in the stash for these, because the ServerVar bits are definitely the area I'm lacking context here and also glad that it already does the right thing